Building: Boost dependency errors

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

```
/compile/freecad/src/Mod/Part/App/Geometry.cpp: In member function ‘void Part::Geometry::createNewTag()’:
/compile/freecad/src/Mod/Part/App/Geometry.cpp:462:19: error: ‘mt19937’ in namespace ‘boost’ does not name a type
  462 |     static boost::mt19937 ran;
      |                   ^~~~~~~
/compile/freecad/src/Mod/Part/App/Geometry.cpp:469:9: error: ‘ran’ was not declared in this scope; did you mean ‘tan’?
  469 |         ran.seed(static_cast<unsigned int>(std::time(nullptr)));
      |         ^~~
      |         tan
/compile/freecad/src/Mod/Part/App/Geometry.cpp:472:56: error: ‘mt19937’ is not a member of ‘boost’; did you mean ‘std::mt19937’?
  472 |     static boost::uuids::basic_random_generator<boost::mt19937> gen(&ran);
      |                                                        ^~~~~~~
In file included from /usr/include/c++/14.2.0/random:48,
                 from /usr/local/include/opencascade/NCollection_UBTreeFiller.hxx:22,
                 from /usr/local/include/opencascade/Extrema_HUBTreeOfSphere.hxx:19,
                 from /usr/local/include/opencascade/Extrema_GenExtPS.hxx:23,
                 from /usr/local/include/opencascade/Extrema_ExtPS.hxx:25,
                 from /usr/local/include/opencascade/GeomPlate_BuildPlateSurface.hxx:30,
                 from /compile/freecad/src/Mod/Part/App/Geometry.h:49,
                 from /compile/freecad/src/Mod/Part/App/Geometry.cpp:119:
/usr/include/c++/14.2.0/bits/random.h:1717:37: note: ‘std::mt19937’ declared here
 1717 |     0xefc60000UL, 18, 1812433253UL> mt19937;
      |                                     ^~~~~~~
/compile/freecad/src/Mod/Part/App/Geometry.cpp:472:63: error: template argument 1 is invalid
  472 |     static boost::uuids::basic_random_generator<boost::mt19937> gen(&ran);
      |                                                               ^
/compile/freecad/src/Mod/Part/App/Geometry.cpp:472:70: error: ‘ran’ was not declared in this scope; did you mean ‘tan’?
  472 |     static boost::uuids::basic_random_generator<boost::mt19937> gen(&ran);
      |                                                                      ^~~
      |                                                                      tan
/compile/freecad/src/Mod/Part/App/Geometry.cpp:474:14: error: ‘gen’ cannot be used as a function
  474 |     tag = gen();
      |           ~~~^~
make[2]: *** [src/Mod/Part/App/CMakeFiles/Part.dir/build.make:3185: src/Mod/Part/App/CMakeFiles/Part.dir/Geometry.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7140: src/Mod/Part/App/CMakeFiles/Part.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

### Full version info

```shell
```


### Subproject(s) affected?

Part

### Anything else?

I fixed the problem. I just apply the proposal of gcc ;)  
i just replace `boost::mt19937` by `std::mt19937` and freecad builds correctly.

Regards,  
Jean-Sebastien Tarot

This the patch! it is a diff

```diff
462c462
<     static std::mt19937 ran;
---
>     static boost::mt19937 ran;
472c472
<     static boost::uuids::basic_random_generator<std::mt19937> gen(&ran);
---
>     static boost::uuids::basic_random_generator<boost::mt19937> gen(&ran);
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/Mod/Part/App/Geometry.cpp
+++ b/src/Mod/Part/App/Geometry.cpp
@@ -102,6 +102,7 @@
 # include <GeomAdaptor_HCurve.hxx>
 # endif
 
+# include <boost/random.hpp>
 # include <cmath>
 # include <ctime>
 #endif //_PreComp_

--- a/src/Mod/Part/App/PreCompiled.h
+++ b/src/Mod/Part/App/PreCompiled.h
@@ -65,6 +65,7 @@
 #include <boost/algorithm/string/predicate.hpp>
 #include <boost/core/ignore_unused.hpp>
 #include <boost/math/special_functions/fpclassify.hpp>
+#include <boost/random.hpp>
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
 

--- a/src/Mod/Sketcher/App/Constraint.cpp
+++ b/src/Mod/Sketcher/App/Constraint.cpp
@@ -23,6 +23,7 @@
 #include "PreCompiled.h"
 #ifndef _PreComp_
 #include <QDateTime>
+#include <boost/random.hpp>
 #include <cmath>
 #endif
 

--- a/src/Mod/Sketcher/App/PreCompiled.h
+++ b/src/Mod/Sketcher/App/PreCompiled.h
@@ -50,6 +50,7 @@
 #include <boost/geometry/geometries/register/point.hpp>
 #include <boost/iostreams/device/array.hpp>
 #include <boost/iostreams/stream.hpp>
+#include <boost/random.hpp>
 #include <boost/range/adaptor/map.hpp>
 #include <boost/uuid/uuid_io.hpp>
 #include <boost_geometry.hpp>

--- a/src/Mod/TechDraw/App/CenterLine.cpp
+++ b/src/Mod/TechDraw/App/CenterLine.cpp
@@ -23,6 +23,7 @@
 
 #include "PreCompiled.h"
 #ifndef _PreComp_
+    #include <boost/random.hpp>
     #include <boost/uuid/uuid_io.hpp>
     #include <boost/uuid/uuid_generators.hpp>
     #include <BRepBuilderAPI_MakeEdge.hxx>

--- a/src/Mod/TechDraw/App/Cosmetic.cpp
+++ b/src/Mod/TechDraw/App/Cosmetic.cpp
@@ -24,6 +24,7 @@
 #include "PreCompiled.h"
 #ifndef _PreComp_
 # include <BRepBuilderAPI_MakeEdge.hxx>
+# include <boost/random.hpp>
 # include <boost/uuid/uuid_generators.hpp>
 # include <boost/uuid/uuid_io.hpp>
 #endif

--- a/src/Mod/TechDraw/App/CosmeticVertex.cpp
+++ b/src/Mod/TechDraw/App/CosmeticVertex.cpp
@@ -25,6 +25,7 @@
 
 #include "PreCompiled.h"
 #ifndef _PreComp_
+    #include <boost/random.hpp>
     #include <boost/uuid/uuid_generators.hpp>
     #include <boost/uuid/uuid_io.hpp>
 #endif // _PreComp_

--- a/src/Mod/TechDraw/App/Geometry.cpp
+++ b/src/Mod/TechDraw/App/Geometry.cpp
@@ -24,6 +24,7 @@
 
 #ifndef _PreComp_
 # include <cmath>
+# include <boost/random.hpp>
 # include <boost/uuid/uuid_generators.hpp>
 # include <boost/uuid/uuid_io.hpp>
 

--- a/src/Mod/TechDraw/App/PreCompiled.h
+++ b/src/Mod/TechDraw/App/PreCompiled.h
@@ -47,6 +47,7 @@
 // boost
 #include <boost/graph/boyer_myrvold_planar_test.hpp>
 #include <boost/graph/is_kuratowski_subgraph.hpp>
+#include <boost/random.hpp>
 #include <boost_regex.hpp>
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_generators.hpp>

